Tasks/SonarQubePostTest/SummaryReport/QualityGateSection.ps1

```diff
@@ -144,20 +145,106 @@ function GetQualityGateWarningsAndErrors
     return $messages
 }
 
-function GetMetricFriendlyName
+
+function GetMetricValueDisplayLabel
+{
+    param ($metricNames, $metricKey, $numericValue)
+    
+    $matchingMetric = $metricNames | Where-Object {$_.key -eq $metricKey}
+    
+    if (!(HasElements $matchingMetric))
+    {
+        Write-Warning "No metric with the key $metricKey found"
+        return $numericValue
+    }
+     
+    if (@($matchingMetric).Count -gt 1) 
+    {
+        Write-Warning "Multiple metrics with the key $metricKey found"
+        return $numericValue     
+    }
+    
+    $type = $matchingMetric.type
+    
+    if ($type -eq "WORK_DUR")
+    {
+        return (GetWorkDurationLabel ($numericValue -as [int]))
+    }  
+    
+    $unit = GetUnitDisplayLabel
+    return ($numericValue + $unit)    
+}
+
+function GetMetricNameDisplayLabel
 {
     param ($metricNames, $metricKey)
+    
     $matchingMetric = $metricNames | Where-Object {$_.key -eq $metricKey} 
     
-    Assert (HasElements $matchingMetric) "No metric with the key $metricKey found"
-    Assert (@($matchingMetric).Count -eq 1) "Multiple metrics with the key $metricKey found"
+    if (!(HasElements $matchingMetric))
+    {
+        Write-Warning "No metric with the key $metricKey found"
+        return $metricKey  
+    } 
+    
+    if (@($matchingMetric).Count -gt 1) 
+    {
+        Write-Warning "Multiple metrics with the key $metricKey found"
+        return $metricKey      
+    }
     
     $name = $matchingMetric.name 
     
     return $name
 }
 
-function GetComparatorDisplayValue
+#
+# SonarQube gives work durations in minutes and it uses complex logic to transform those values to hours, work days, weeks, months etc. 
+# At this point we only show hours and minutes.
+#
+function GetWorkDurationLabel
```

Comment:
Could you not use `System.TimeSpan` and then use the various hour, min, sec properties?